RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreationAlgo.cc

```diff
@@ -472,72 +509,78 @@ void CaloTowersCreationAlgo::assignHitHcal(const CaloRecHit * recHit) {
   // SPECIAL handling of tower 28 merged depths --> half into tower 28 and half into tower 29
   if (detId.det()==DetId::Hcal && 
       HcalDetId(detId).subdet()==HcalEndcap &&
-	  (theHcalPhase==0 || theHcalPhase==1) &&
-	  std::find(mergedDepths.begin(), mergedDepths.end(), HcalDetId(detId).depth())!=mergedDepths.end() &&
+      (theHcalPhase==0 || theHcalPhase==1) &&
       //HcalDetId(detId).depth()==3 &&
       HcalDetId(detId).ietaAbs()==theHcalTopology->lastHERing()-1) {
-
-    //////////////////////////////    unsigned int chStatusForCT = hcalChanStatusForCaloTower(recHit);
-      
-    // bad channels are counted regardless of energy threshold
-
-    if (chStatusForCT == CaloTowersCreationAlgo::BadChan) {
-      CaloTowerDetId towerDetId = theTowerConstituentsMap->towerOf(detId);
-      if (towerDetId.null()) return;
-      MetaTower & tower28 = find(towerDetId);
-      CaloTowerDetId towerDetId29(towerDetId.ieta()+towerDetId.zside(),
-                                  towerDetId.iphi());
-      MetaTower & tower29 = find(towerDetId29);
-      tower28.numBadHcalCells += 1;
-      tower29.numBadHcalCells += 1;
+    bool merge(false);
+    if (subdetOne == HcalEndcap && (std::find(phizOne.begin(),phizOne.end(),std::pair<int,int>(HcalDetId(detId).iphi(),HcalDetId(detId).zside())) == phizOne.end())) {
+      merge = (std::find(mergedDepthsOne.begin(), mergedDepthsOne.end(), HcalDetId(detId).depth())!=mergedDepthsOne.end());
```

Comment:
I just wanted to double-check, since this piece looks confusing.

At https://github.com/cms-sw/cmssw/pull/17467/files#diff-161bc85b60d87e850649585423571364R303
the mergedDepthsOne is filled from {zside, iphi} __from__ phizOne[0].
Here the "merge" boolean is set only for elements which are __not from__ phizOne.
Is this the correct choice?